source/common/grpc/google_async_client_impl.cc

```diff
@@ -10,18 +10,56 @@
 namespace Envoy {
 namespace Grpc {
 
+GoogleAsyncClientThreadLocal::GoogleAsyncClientThreadLocal()
+    : completion_thread_(new Thread::Thread([this] { completionThread(); })) {}
+
+GoogleAsyncClientThreadLocal::~GoogleAsyncClientThreadLocal() {
+  // Force streams to shutdown and invoke TryCancel() to start the drain of
+  // pending op. If we don't do this, Shutdown() below can jam on pending ops.
+  // This is also required to satisfy the contract that once Shutdown is called,
+  // streams no longer queue any additional tags.
+  for (auto it = streams_.begin(); it != streams_.end();) {
+    (*it++)->resetStream();
```

Comment:
Any particular reason why you haven't written this as `(*it)->resetStream()` and done the iterator increment in the for loop params?

Reply:
Added comment; it has to do with resetStream() performing a potential erase(), which would invalidate the current iterator.

Reply:
Awesome. I've fixed at least one bug of that sort in Envoy, so I appreciate both that you got it right and the comment.